File: 0334_increasing_triplet_subsequence/solution.rb
# LeetCode 0334 - Increasing Triplet Subsequence
# https://leetcode.com/problems/increasing-triplet-subsequence/

class Solution
  def increasing_triplet(nums)
    first = second = Float::INFINITY
    nums.each do |num|
      if num <= first
        first = num
      elsif num <= second
        second = num
      else
        return true
      end
    end
    false
  end

  alias_method :increasingTriplet, :increasing_triplet
end

File: 0334_increasing_triplet_subsequence/solution.rs
// LeetCode 0334 - Increasing Triplet Subsequence
// https://leetcode.com/problems/increasing-triplet-subsequence/

impl Solution {
    pub fn increasing_triplet(nums: Vec<i32>) -> bool {
        let mut first = i32::MAX;
        let mut second = i32::MAX;
        for num in nums {
            if num <= first {
                first = num;
            } else if num <= second {
                second = num;
            } else {
                return true;
            }
        }
        false
    }
}
